store/core.go

```diff
@@ -614,31 +614,116 @@ func (s *store) GetPropertyLabels(
 func (s *store) btGetTriples(
 	ctx context.Context, in *pb.GetTriplesRequest, out *pb.GetTriplesResponse) error {
 	dcids := in.GetDcids()
-
-	rowList := bigtable.RowList{}
+	var regDcids, obsDcids []string
 	for _, dcid := range dcids {
-		rowList = append(rowList, fmt.Sprintf("%s%s", util.BtTriplesPrefix, dcid))
+		if strings.HasPrefix(dcid, "dc/o/") {
+			obsDcids = append(obsDcids, dcid)
+		} else {
+			regDcids = append(regDcids, dcid)
+		}
 	}
 
 	resultsMap := map[string][]*Triple{}
-	if err := bigTableReadRowsParallel(ctx, s.btTable, rowList,
-		func(btRow bigtable.Row) error {
-			// Extract DCID from row key.
-			dcid := strings.TrimPrefix(btRow.Key(), util.BtTriplesPrefix)
 
-			if len(btRow[util.BtFamily]) > 0 {
-				btRawValue := btRow[util.BtFamily][0].Value
-				btJSONRaw, err := util.UnzipAndDecode(string(btRawValue))
-				if err != nil {
-					return err
+	// Regular DCIDs.
+	if len(regDcids) > 0 {
+		rowList := bigtable.RowList{}
+		for _, dcid := range regDcids {
+			rowList = append(rowList, fmt.Sprintf("%s%s", util.BtTriplesPrefix, dcid))
+		}
+		if err := bigTableReadRowsParallel(ctx, s.btTable, rowList,
+			func(btRow bigtable.Row) error {
+				// Extract DCID from row key.
+				dcid := strings.TrimPrefix(btRow.Key(), util.BtTriplesPrefix)
+
+				if len(btRow[util.BtFamily]) > 0 {
+					btRawValue := btRow[util.BtFamily][0].Value
+					btJSONRaw, err := util.UnzipAndDecode(string(btRawValue))
+					if err != nil {
+						return err
+					}
+					var btTriples TriplesCache
+					json.Unmarshal(btJSONRaw, &btTriples)
+					resultsMap[dcid] = filterTriplesLimit(dcid, btTriples.Triples, int(in.GetLimit()))
 				}
-				var btTriples TriplesCache
-				json.Unmarshal(btJSONRaw, &btTriples)
-				resultsMap[dcid] = filterTriplesLimit(dcid, btTriples.Triples, int(in.GetLimit()))
+				return nil
+			}); err != nil {
+			return err
+		}
+	}
+
+	objPlaceIDNameMap := map[string]string{}
+	if len(obsDcids) > 0 {
+		// Observation DCIDs.
+		obsRowList := bigtable.RowList{}
+		for _, dcid := range obsDcids {
+			for _, pred := range []string{"0", "1"} {
+				obsRowList = append(obsRowList,
+					fmt.Sprintf("%s%s-%s", util.BtObsAncestorPrefix, dcid, pred))
 			}
-			return nil
-		}); err != nil {
-		return err
+		}
+		if err := bigTableReadRowsParallel(ctx, s.btTable, obsRowList,
+			func(btRow bigtable.Row) error {
+				// Extract DCID from row key.
+				parts := strings.Split(btRow.Key(), "-")
+				dcid := strings.TrimPrefix(parts[0], util.BtObsAncestorPrefix)
+				var pred string
+				if parts[1] == "0" {
```

Comment:
Echoing Shanth's comment above, maybe change "0" and "1" to use constants

Reply:
Done.